interfaces/acados_template/acados_template/acados_ocp_solver.py

```diff
@@ -96,6 +81,291 @@ def make_ocp_dims_consistent(acados_ocp):
         dims.np = 0
     else:
         dims.np = casadi_length(model.p)
+    if acados_ocp.parameter_values.shape[0] != dims.np:
+        raise Exception('inconsistent dimension np, regarding model.p and parameter_values.')
+
+    ## cost
+    # path
+    if cost.cost_type == 'LINEAR_LS':
+        ny = cost.W.shape[0]
+        if cost.Vx.shape[0] != ny or cost.Vu.shape[0] != ny:
+            raise Exception('inconsistent dimension ny, regarding W, Vx, Vu.')
```

Comment:
How about printing the actual dimensions that are causing the Exception? Given that the dimensions are now detected and not set by the user, I guess you would need to print something like 

`'[...] W[ny,ny], Vx[ny,nx], Vu[ny,nu], you have: W[{},{}], Vx[{},{}], Vu[{},{}]'.format( [...]` 

I think it would improve a lot user friendliness :)